appdatabase/database.go

```diff
@@ -198,11 +188,144 @@ func MigrateEnsUsernames(db *sql.DB) error {
+		ALTER TABLE transfers ADD COLUMN contract_address TEXT;
+		ALTER TABLE transfers ADD COLUMN gas_used INT;
+		ALTER TABLE transfers ADD COLUMN tx_index INT;
+
+		ALTER TABLE transfers ADD COLUMN protected BOOLEAN;
+		ALTER TABLE transfers ADD COLUMN gas_limit UNSIGNED INT;
+		ALTER TABLE transfers ADD COLUMN gas_price_clamped64 INT;
+		ALTER TABLE transfers ADD COLUMN gas_tip_cap_clamped64 INT;
+		ALTER TABLE transfers ADD COLUMN gas_fee_cap_clamped64 INT;
+		ALTER TABLE transfers ADD COLUMN amount_padded128hex CHAR(32);
+		ALTER TABLE transfers ADD COLUMN account_nonce INT;
+		ALTER TABLE transfers ADD COLUMN size INT;
+
+		CREATE INDEX transfers_status ON transfers (status);`
+
+	rowIndex := 0
+	mightHaveRows := true
+
+	_, err := sqlTx.Exec(newColumnsAndIndexSetup)
+	if err != nil {
+		return err
+	}
+
+	for mightHaveRows {
+		var chainID uint64
+		var hash common.Hash
+		var address common.Address
+
+		rows, err := sqlTx.Query(`SELECT hash, address, network_id, tx, receipt FROM transfers WHERE tx IS NOT NULL OR receipt IS NOT NULL LIMIT ? OFFSET ?`, batchSize, rowIndex)
+		if err != nil {
+			return err
+		}
+
+		curProcessed := 0
+		for rows.Next() {
+			tx := &types.Transaction{}
+			r := &types.Receipt{}
+
+			// Scan row data into the transaction and receipt objects
+			nullableTx := sqlite.JSONBlob{Data: tx}
+			nullableR := sqlite.JSONBlob{Data: r}
+			err = rows.Scan(&hash, &address, &chainID, &nullableTx, &nullableR)
+			if err != nil {
+				rows.Close()
+				return err
+			}
+
+			var currentRow []interface{}
+			// Check if the receipt is not null before transferring the receipt data
+			if nullableR.Valid {
+				currentRow = append(currentRow, r.Status, r.Type, r.CumulativeGasUsed, r.ContractAddress, r.GasUsed, r.TransactionIndex)
+			} else {
+				for i := 0; i < 6; i++ {
+					currentRow = append(currentRow, nil)
+				}
+			}
+
+			if nullableTx.Valid {
+				gasPrice := sqlite.BigIntToClampedInt64(tx.GasPrice())
+				gasTipCap := sqlite.BigIntToClampedInt64(tx.GasTipCap())
+				gasFeeCap := sqlite.BigIntToClampedInt64(tx.GasFeeCap())
+				value := sqlite.BigIntToPadded128BitsStr(tx.Value())
+
+				currentRow = append(currentRow, tx.Protected(), tx.Gas(), gasPrice, gasTipCap, gasFeeCap, value, tx.Nonce(), int64(tx.Size()))
+			} else {
+				for i := 0; i < 8; i++ {
+					currentRow = append(currentRow, nil)
+				}
+			}
+			currentRow = append(currentRow, hash, address, chainID)
+			batchEntries = append(batchEntries, currentRow)
+
+			curProcessed++
+		}
+		rowIndex += curProcessed
+
+		// Check if there was an error in the last rows.Next()
+		rows.Close()
+		if err = rows.Err(); err != nil {
+			return err
+		}
+		mightHaveRows = (curProcessed == batchSize)
+
+		// insert extracted data into the new columns
+		if len(batchEntries) > 0 {
+			var stmt *sql.Stmt
+			stmt, err = sqlTx.Prepare(`UPDATE transfers SET status = ?, receipt_type = ?, cumulative_gas_used = ?, contract_address = ?, gas_used = ?, tx_index = ?,
```

Comment:
not sure if by `receipt_type` you meant "type obtained from the receipt" or "type of receipt", I think the proper name for this is "Transaction Type" (https://docs.infura.io/infura/networks/ethereum/concepts/transaction-type)